Xamarin.Forms.Platform.Android/Renderers/RefreshViewRenderer.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System;
+using System.ComponentModel;
+using Android.Content;
+using Android.OS;
+using Android.Support.V4.View;
+using Android.Support.V4.Widget;
+using Android.Support.V7.Widget;
+using Android.Views;
+using Android.Widget;
+using Xamarin.Forms.Internals;
+using AView = Android.Views.View;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	public class RefreshViewRenderer : SwipeRefreshLayout, IVisualElementRenderer, IEffectControlProvider, SwipeRefreshLayout.IOnRefreshListener
+	{
+		bool _isDisposed;
+		bool _init;
+		bool _refreshing;
+		IVisualElementRenderer _renderer;
+		int? _defaultLabelFor;
+
+		public event EventHandler<VisualElementChangedEventArgs> ElementChanged;
+		public event EventHandler<PropertyChangedEventArgs> ElementPropertyChanged;
+
+		public RefreshViewRenderer(Context context)
+			: base(context)
+		{
+
+		}
+
+		public VisualElementTracker Tracker { get; private set; }
+
+		public ViewGroup ViewGroup => this;
+
+		public AView View => this;
+
+		public SwipeRefreshLayout SwipeRefreshLayout => View as SwipeRefreshLayout;
+
+		public VisualElement Element { get; private set; }
+
+		public RefreshView RefreshView => Element != null ? (RefreshView)Element : null;
+
+		public override bool Refreshing
+		{
+			get
+			{
+				return _refreshing;
+			}
+			set
+			{
+				_refreshing = value;
+
+				if (RefreshView != null && RefreshView.IsRefreshing != _refreshing)
+					RefreshView.IsRefreshing = _refreshing;
+
+				if (base.Refreshing == _refreshing)
+					return;
+
+				base.Refreshing = _refreshing;
```

Comment:
To make this simpler, can't we do the following?

```
if(RefreshView != null)
   RefreshView.IsRefreshing = _refreshing;

base.Refreshing = _refreshing;
```

In other words, the extra equality checks may be unnecessary since setting a property to its current value should have minimal impact, no?